ocf_datapipes/load/pv/utils.py

```diff
@@ -111,3 +112,38 @@ def encode_label(indexes: List[str], label: str) -> List[str]:
     new_index = [int(int(col) * 10 + label_index) for col in indexes]
 
     return new_index
+
+
+def xr_to_df(pv_power: xr.Dataset, ssid: str, date_oi: str) -> pd.DataFrame():
+    """Function that gives a pv system df of a single day with its output
+
+    Converts xarray dataset into a pandas dataframe,
+    and its values for a single pv systme and a single day
+
+    Args
+        pv_power: pv.netcdf file which contains system is and datetime
+                    with corresponding pv output values
+        ssid: A single ID of the pv system
+        date_oi: Date of Interest
+    """
+    date_1 = datetime.strptime(date_oi, "%Y-%m-%d")
+    on_pv_system = pv_power[ssid].to_dataframe()
+    next_day = date_1 + timedelta(days=1)
+    on_pv_system = on_pv_system[(on_pv_system.index < next_day) & (on_pv_system.index > date_1)]
+    return on_pv_system
+
+
+def dates_list(pv_power: xr.Dataset) -> List:
+    """Provides a list of all the dates in the xr dataset
+
+    Converts dates as coordinates from xarray dataset to a list
```

Comment:
I think the doc/function name is a bit misleading, here you are returning a list of strings that look like dates, not actual dates.

Reply:
Yes, I still need to tidy up the comments and the code as well, I have been learning through `torch datapipes` and `ocf_datapipes` docs, I think I have a good understanding now, I'll redo the stuff to adapt my code to the ocf_datapipes!